### ISSUE TITLE
sokol_audio.h: error building for Android, linker error (undefined reference)

### ISSUE THREAD

**Issue by Velocity-** [floooh/sokol#250]
When building the Android examples, `sokol_audio.h` fails to build:

```
> ./fips build | grep error
./sokol/sokol_audio.h:1540: error: undefined reference to 'slCreateEngine'
./sokol/sokol_audio.h:0: error: undefined reference to 'SL_IID_ENGINE'
./sokol/sokol_audio.h:0: error: undefined reference to 'SL_IID_VOLUME'
./sokol/sokol_audio.h:0: error: undefined reference to 'SL_IID_VOLUME'
./sokol/sokol_audio.h:0: error: undefined reference to 'SL_IID_VOLUME'
./sokol/sokol_audio.h:0: error: undefined reference to 'SL_IID_ANDROIDSIMPLEBUFFERQUEUE'
./sokol/sokol_audio.h:0: error: undefined reference to 'SL_IID_PLAY'
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
```

Environment:
MacBook Pro 2019, OSX Catalina

**Comment by floooh** [comment]
Looks like one or some of the build targets don't link with OpenSLES. I'll have a look (installing the Android SDK on Catalina takes unusually long while I'm typing this reply).

**Comment by floooh** [comment]
Ok, it was the 'cpp-compile-test' target, fixed in b51db43d81ffe2d3ad14d99c0a10b53f0c2b8315, there was also a missing type case in sokol_audio.h when compiled as C++: https://github.com/floooh/sokol-samples/commit/b51db43d81ffe2d3ad14d99c0a10b53f0c2b8315

I'll try to add Android to the automated CI tests to catch such errors in the future (hopefully installing the Android SDK/NDK won't run into timeouts)

**Comment by floooh** [comment]
Ok, I have added an Android compilation check to github actions, both in the sokol, and sokol-samples repository. This should catch such issues in the future.

https://github.com/floooh/sokol-samples/runs/397580444
